File: Medium/Maximum Bags With Full Capacity of Rocks/maximum-bags-with-full-capacity-of-rocks.py
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:

        diff = [ capacity[i] - rocks[i] for i in range(len(capacity)) ]
        diff.sort()

        res = 0

        for i in range(len(diff)):
            if diff[i] == 0:
                res += 1
                continue
            additionalRocks -= diff[i]
            if additionalRocks >= 0:
                res += 1
        
        return res

File: Medium/Maximum Bags With Full Capacity of Rocks/maximum-bags-with-full-capacity-of-rocks.cpp
class Solution {
public:
    int maximumBags(vector<int>& capacity, vector<int>& rocks, long additionalRocks) {
        vector<int> diff;

        for (int i = 0; i < capacity.size(); ++i)
            diff.emplace_back(capacity[i] - rocks[i]);
        
        sort(diff.begin(), diff.end());
        int res = 0;

        for (auto i : diff) {
            if (i == 0) {
                res += 1;
                continue;
            }
            additionalRocks -= i;
            if (additionalRocks >= 0)
                res += 1;
        }

        return res;
    }
};
